Formula/m/mit-scheme.rb

```diff
@@ -55,11 +49,17 @@ def install
     # not the correct type." Only Haswell and above appear to be impacted.
     # Reported 23rd Apr 2016: https://savannah.gnu.org/bugs/index.php?47767
     # NOTE: `unless build.bottle?` avoids overriding --bottle-arch=[...].
+
     ENV["HOMEBREW_OPTFLAGS"] = "-march=#{Hardware.oldest_cpu}" unless build.bottle?
 
     resource("bootstrap").stage do
       cd "src"
-      system "./configure", "--prefix=#{buildpath}/staging", "--without-x"
+      if OS.mac?
+        system "./configure", "--prefix=#{buildpath}/staging", "--without-x", \
+          "CC=/usr/bin/gcc", "CXX=/usr/bin/g++"
```

Comment:
`/usr/bin/gcc` points to clang, which is already the default.

Reply:
Yes, but after lots of trial and error, it seems that this is the only way that it compiles when using `homebrew`. When executing the configuration script to generate the Makefile in the terminal, it works normally. However, when doing it through `homebrew` it throws many errors which are only solved when specifying `/usr/bin/gcc` and `/usr/bin/g++`. Other methods and compilers do not seem to work. This quirk is only specific to macOS.

Reply:
Let's add it at the top of the formula as `ENV["CC"] = "/usr/bin/gcc"` then.

Reply:
I think we should do that for now, we can always override the check.

Reply:
> I think we should do that for now, we can always override the check.

I think this will fail the tap syntax check, which will fail all Homebrew/core PRs.

Reply:
Alright, then this should work, although flagged by `brew audit`:

```rb
if OS.mac?
  if MacOS.version < :monterey && Hardware::CPU.intel?
    ENV["HOMEBREW_OPTFLAGS"] = "-march=#{Hardware.oldest_cpu}" unless build.bottle?
  end
  ENV["CC"] = "/usr/bin/gcc"
  ENV["CXX"] = "/usr/bin/g++"
  ENV["OBJC"] = "/usr/bin/gcc"
  ENV["OBJCXX"] = "/usr/bin/g++"
end

system "./configure", "--prefix=#{buildpath}/staging", "--without-x"
```

Reply:
What is the error if you drop setting `CC`, `CXX`, etc?

Reply:
It seems that Homebrew by default passes some settings when using `clang` which results in the `configure` script from `mit-scheme` reporting that `clang` does not produce a C executable. If I try switching to `gcc` installed via Homebrew then, regardless of version, it throws an error relating to the `microcode` check from the script, thus once more failing the build. 

When I just extract the tarball and run the `configure` script normally from macOS Terminal, it builds normally. But Homebrew causes these errors that are only resolved when manually specifying it should use `/usr/bin/gcc` and `/usr/bin/g++`. 

That is why I committed `system "./configure", "--prefix=#{buildpath}/staging", "--without-x", \ "CC=/usr/bin/gcc", "CXX=/usr/bin/g++"` as it resolves the issue while still complying with `brew audit --strict`.

Reply:
> It seems that Homebrew by default passes some settings when using `clang` which results in the `configure` script from `mit-scheme` reporting that `clang` does not produce a C executable.

Can you upload the resulting `config.log` from this failure?

Reply:
> > It seems that Homebrew by default passes some settings when using `clang` which results in the `configure` script from `mit-scheme` reporting that `clang` does not produce a C executable.
> 
> Can you upload the resulting `config.log` from this failure?

I tried running `HOMEBREW_NO_INSTALL_FROM_API=1 HOMEBREW_KEEP_TMP=1 brew install --build-from-source mit-scheme` but it still destroys the whole tarball and logs when exiting upon fail with the following output:

```
./configure
--prefix=/private/tmp/mit-scheme-20240617-55644-k84v4j/mit-scheme-12.1/staging
--without-x

checking whether make sets $(MAKE)... yes
checking build system type... arm-apple-darwin23.5.0
checking host system type... arm-apple-darwin23.5.0
checking for gcc... clang
checking whether the C compiler works... no
configure: error: in `/private/tmp/mit-scheme--bootstrap-20240617-55644-tiqpni/mit-scheme-12.1/src':
configure: error: C compiler cannot create executables
See `config.log' for more details
```

**However,** I noticed something very interesting: the issue seems to be caused by this line:
```
ENV["HOMEBREW_OPTFLAGS"] = "-march=#{Hardware.oldest_cpu}" unless build.bottle?
```
When I comment it out, the build runs perfectly fine on macOS Sonoma on Apple Silicon, even when removing the `ENV["CC"]` and `ENV["CXX"]` hardcodings. Thus, removing all need for them. I can still preserve that line in case it is needed for older hardware as follows:
```rb
if OS.mac? && MacOS.version < :monterey && Hardware::CPU.intel?
  ENV["HOMEBREW_OPTFLAGS"] = "-march=#{Hardware.oldest_cpu}" unless build.bottle?
  end
```

I think we should make a new commit which only keeps adds this part (along with the switch to the sv1 tarball), thus eliminating the need to specify / hardcode `/usr/bin/gcc` in the formula.

Reply:
What's special about Monterey? The reason for that line is specifically because `-march=native` fails to build on the latest Intel hardware due to an upstream problem. So either it should be gated by `Hardware::CPU.intel?` or upstream has fixed the problem and it can be removed entirely.

Reply:
I've removed it and all the CI/CD builds have passed so it no longer seems necessary to include that line. 

Thank you all for your help in this matter.